File: API/Entities/Post.cs
using System;
using System.Collections.Generic;

namespace API.Entities
{
    public class Post
    {
        public int Id { get; set; }

        public string Content { get; set; } = null!;

        public DateTime LastUpdatedAt { get; set; } = DateTime.UtcNow;

        public string AuthorId { get; set; } = null!;
        public ApplicationUser Author { get; set; } = null!;

        public ICollection<PostMedia> Media { get; set; } = [];
        public ICollection<PostComment> Comments { get; set; } = [];
        public ICollection<Like> Likes { get; set; } = [];
    }
}

File: client/src/app/shared/Contracts/Post.ts
import { PostComment } from "./PostComment";

export interface Post {
    id: number;
    content: string;
    authorId: string;
    authorName: string;
    authorImageUrl: string;
    mediaUrl: string;
    comments: PostComment[];
    likes: number;
    isLiked: boolean;
}

